Boundary "loop" selection + selection state cleanup

MightyBOBcnc opened the pull request:
Hi, me again.  I added a new function for selecting a boundary edge "loop" of connected edges like Maya's selection of a mesh's Border Edges.  It only selects the "loop" that is connected to the starting edge, not every boundary edge on the mesh.

![](https://i.imgur.com/B2rdlY8.gif)

There are two (three?) design questions that maybe should be answered. 
1: Should there be a Preference to enable/disable this?  I can see how some users would prefer for the boundary edge selection to terminate at corners or 'poles' where more than 2 faces connect at a boundary vertex but if the goal is to emulate Maya, well Maya doesn't terminate; it grabs the full border edge "loop".
2: The way the function is implemented right now it will only select up to 1000 edges in a boundary edge loop which is kind of a performance/safety failsafe.  Do we think that is okay or should we make it unlimited and let the user take responsibility for any performance issues?
2a: If we think a limit is sensible for performance reasons, should that be configurable in a preference or left at 1000?

-----
I also added a couple more selection flushes to keep the selection state 'happy' the way Blender likes, updated the author line, and the version number.

MightyBOBcnc committed:
Merge pull request #1 from Stromberg90/master

Update to latest

MightyBOBcnc committed:
Flush selection state; new boundary loop selection

- Added bm.select_flush_mode() at end of the 3 main defs (vert, face, edge selection) to update/maintain the selection state.  https://docs.blender.org/api/current/bmesh.html#keeping-a-correct-state  
- Added new function get_boundary_edge_loop which allows to double click / shift+double click on a boundary edge (hole in the mesh) and select all of the boundary edges that are connected/contiguous (like Maya's Border Edge selection).  
![](https://i.imgur.com/B2rdlY8.gif)

Stromberg90 commented:
I'm happy to see this getting improved :)
I don't think we need a preference for this, since this is how it's supposed to work, however if people complain about it later, then yeah we can add that.
Also think we can get rid off the edge limit, and if it becomes a problem see if we can speed it up, I feel a limit might make people(myself included) think that it's broken or something.

MightyBOBcnc committed:
Remove boundary loop limit

- Removed limit of 1000 edge selection for boundary edge "loops".
- Removed a boundary check that really wouldn't have saved you anyway if somehow you passed a non-boundary edge to get_boundary_edges.  Make sure you do boundary check BEFORE passing an edge to get_boundary_edges.

MightyBOBcnc commented:
Alright, edge limit removed.  Should be green for testing and merge.
